server.go

```diff
@@ -140,6 +140,34 @@ func (s *Server) handleConn(conn net.Conn) {
 					break
 				}
 				writeErr = writer.WriteObjects(stats.toRedis()...)
+			// Get all topics available
+			//
+			// KEYS
+			case "KEYS":
+				// Normally in redis proto KEYS has an argument but we ingore it.
+				_ = string(command.Get(1))
+
+				prod, err := c.Producer(cfg.Librdkafka.Producer)
+				if err != nil {
+					writeErr = writer.WriteError("ERR Error spawning producer: " + err.Error())
+					break
+				}
+
+				metadata, err := prod.rdProd.GetMetadata(nil, true, 100)
+				if err != nil {
+					writeErr = writer.WriteError("ERR Error getting metadata: " + err.Error())
+					break
+				}
+
+				var topic_names []interface{}
```

Comment:
It returns:
```
./server.go:169:36: cannot use topic_names (type []string) as type []interface {} in argument to writer.WriteObjects
```

I think it has to do with the internal implementation of interfaces, the only thing I found is [this article](http://jordanorelli.com/post/32665860244/how-to-use-interfaces-in-go) and the relevant part is:
```
By running this, you can see that we encounter the following error: cannot use names (type []string) as type []interface {} in function argument. If we want to actually make that work, we would have to convert the []string to an []interface{}:

package main

import (
    "fmt"
)

func PrintAll(vals []interface{}) {
    for _, val := range vals {
        fmt.Println(val)
    }
}

func main() {
    names := []string{"stanley", "david", "oscar"}
    vals := make([]interface{}, len(names))
    for i, v := range names {
        vals[i] = v
    }
    PrintAll(vals)
}
Run it here:http://play.golang.org/p/Dhg1YS6BJS

That’s pretty ugly, but c'est la vie. Not everything is perfect. (in reality, this doesn’t come up very often, because []interface{} turns out to be less useful than you would initially expect)
```

At first I had the topic_names` var as a slice of strings as you say, and has something to "unwrap" the strings and convert them to interfaces but we discussed it a bit with @dtheodor and we converged on just using an interface slice.

Feel free to clarify it more for me, and/or suggest anything else, cause I'm a beginner in go :)

Reply:
Ah I see. Fair enough.